Rebuilding project with errors reports cannot find .ts files

# 🐞 Bug report

### Command (mark with an `x`)
<!-- Can you pin-point the command or commands that are effected by this bug? -->
<!-- ✍️edit: -->
```
- [ ] new
- [X] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?
Yes
<!-- Did this behavior use to work in the previous version? -->
7.x


### Description

With a project that has build errors (after updating to V8), with `ng serve` running, correcting one error and saving the file causes the project to rebuild, but report that it cannot find the main.ts files:

```
ERROR in ./src/main.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/index.js):
Error: /Users/john/Dev/Javascript/Angular/FRC-TimeTracker/src/main.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
    at AngularCompilerPlugin.getCompiledFile (/Users/john/Dev/Javascript/Angular/FRC-TimeTracker/node_modules/@ngtools/webpack/src/angular_compiler_plugin.js:834:23)
    at /Users/john/Dev/Javascript/Angular/FRC-TimeTracker/node_modules/@ngtools/webpack/src/loader.js:41:31
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
ERROR in ./src/polyfills.ts
Module build failed (from ./node_modules/@ngtools/webpack/src/index.js):
Error: /Users/john/Dev/Javascript/Angular/FRC-TimeTracker/src/polyfills.ts is missing from the TypeScript compilation. Please make sure it is in your tsconfig via the 'files' or 'include' property.
    at AngularCompilerPlugin.getCompiledFile (/Users/john/Dev/Javascript/Angular/FRC-TimeTracker/node_modules/@ngtools/webpack/src/angular_compiler_plugin.js:834:23)
    at /Users/john/Dev/Javascript/Angular/FRC-TimeTracker/node_modules/@ngtools/webpack/src/loader.js:41:31
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
ℹ ｢wdm｣: Failed to compile.
    
```

After the first `ng serve`other than the build errors (below), the above errors do not happen.

My build errors
```
   ERROR in src/app/apparel/add-apparel/add-apparel.component.ts(25,4): error TS2554: Expected 2 arguments, but got 1.
    src/app/events/add-edit-event.component.ts(23,4): error TS2554: Expected 2 arguments, but got 1.
    src/app/message/edit-message/edit-message.component.ts(25,4): error TS2554: Expected 2 arguments, but got 1.
    src/app/message/message-list/message-list.component.ts(18,4): error TS2554: Expected 2 arguments, but got 1.
    src/app/reports/student-hours-report.component.ts(21,4): error TS2554: Expected 2 arguments, but got 1.
    src/app/shared/alert-message/alert-message.component.ts(9,4): error TS2554: Expected 2 arguments, but got 1.
    src/app/shared/image-upload/image-upload-service.service.ts(2,22): error TS2307: Cannot find module '@angular/http'.
    src/app/student/student-list/student-list.component.ts(21,4): error TS2554: Expected 2 arguments, but got 1.
```

## 🔬 Minimal Reproduction


## 🔥 Exception or Error
Shown above.


## 🌍 Your Environment
<pre><code>
ng version

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 8.0.1
Node: 12.3.1
OS: darwin x64
Angular: 8.0.0
... animations, cdk, common, compiler, compiler-cli, core, forms
... language-service, material, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.800.1
@angular-devkit/build-angular     0.800.1
@angular-devkit/build-optimizer   0.800.1
@angular-devkit/build-webpack     0.800.1
@angular-devkit/core              8.0.1
@angular-devkit/schematics        8.0.1
@angular/cli                      8.0.1
@ngtools/webpack                  8.0.1
@schematics/angular               8.0.1
@schematics/update                0.800.1
rxjs                              6.5.2
typescript                        3.4.5
webpack                           4.30.0

</code></pre>

--- a/packages/angular_devkit/build_angular/test/browser/rebuild_spec_large.ts
+++ b/packages/angular_devkit/build_angular/test/browser/rebuild_spec_large.ts
@@ -9,7 +9,7 @@
 
 import { Architect } from '@angular-devkit/architect';
 import { TestLogger } from '@angular-devkit/architect/testing';
-import { normalize, virtualFs } from '@angular-devkit/core';
+import { logging, normalize, virtualFs } from '@angular-devkit/core';
 import { debounceTime, take, takeWhile, tap } from 'rxjs/operators';
 import { createArchitect, host, lazyModuleFiles, lazyModuleStringImport } from '../utils';
 
@@ -68,14 +68,11 @@ describe('Browser Builder rebuilds', () => {
 
     const overrides = { watch: true };
 
-    let buildCount = 0;
     let phase = 1;
     const run = await architect.scheduleTarget(target, overrides);
     await run.output.pipe(
       tap(result => {
         expect(result.success).toBe(true, 'build should succeed');
-        buildCount++;
-
         const hasLazyChunk = host.scopedSync().exists(normalize('dist/lazy-lazy-module.js'));
         switch (phase) {
           case 1:
@@ -203,6 +200,47 @@ describe('Browser Builder rebuilds', () => {
       tap(() => host.writeMultipleFiles({ 'src/type.ts': `export type MyType = string;` })),
       take(2),
     ).toPromise();
+  });
+
+  it('rebuilds after errors in JIT', async () => {
+    const origContent = virtualFs.fileBufferToString(
+      host.scopedSync().read(normalize('src/app/app.component.ts')));
+    host.appendToFile('./src/app/app.component.ts', `console.logg('error')`);
+
+    const overrides = { watch: true, aot: false };
+    let buildNumber = 0;
+    const logger = new logging.Logger('');
+    let logs: string[] = [];
+    logger.subscribe(e => logs.push(e.message));
+
+    const run = await architect.scheduleTarget(target, overrides, { logger });
+    await run.output.pipe(
+      debounceTime(1000),
+      tap((buildEvent) => {
+        buildNumber ++;
+        switch (buildNumber) {
+          case 1:
+            // The first build should error out with an error.
+            expect(buildEvent.success).toBe(false);
+            expect(logs.join()).toContain(`Property 'logg' does not exist on type 'Console'`);
+            logs = [];
+            // Fix the error.
+            host.writeMultipleFiles({ 'src/app/app.component.ts': `
+              ${origContent}
+              console.errorr('error');
+            `});
+            break;
+
+          case 2:
+            // The second build should have everything fixed.
+            expect(buildEvent.success).toBe(true);
+            expect(logs.join()).not.toContain('Module build failed');
+            break;
+        }
+      }),
+      take(2),
+    ).toPromise();
+    await run.stop();
   });
 
   it('rebuilds after errors in AOT', async () => {

--- a/packages/ngtools/webpack/src/angular_compiler_plugin.ts
+++ b/packages/ngtools/webpack/src/angular_compiler_plugin.ts
@@ -1195,7 +1195,7 @@ export class AngularCompilerPlugin {
           'AngularCompilerPlugin._emit.ts', diagMode));
 
         if (!hasErrors(allDiagnostics)) {
-          if (this._firstRun || changedTsFiles.size > 20) {
+          if (this._firstRun || changedTsFiles.size > 20 || this._emitSkipped) {
             emitResult = tsProgram.emit(
               undefined,
               undefined,

